clang/include/clang/CConv/ConstraintResolver.h

```diff
@@ -69,9 +70,7 @@ class ConstraintResolver {
 
   CVarSet getAllSubExprConstraintVars(std::vector<Expr *> &Exprs);
   CVarSet getBaseVarPVConstraint(DeclRefExpr *Decl);
-  bool hasPersistentConstraints(clang::Expr *E);
-  CVarSet getPersistentConstraints(clang::Expr *E);
-  void storePersistentConstraints(clang::Expr *E, CVarSet &Vars);
```

Comment:
This was the thing you observed at some point that the `persisted` flag is residual and not needed?

Reply:
Or, I'm seeing later, that perhaps these methods were just moved to `ProgramInfo` ?

Reply:
Moved to `ProgramInfo`. I haven't got rid of persited yet, but I think we can.